ai/src/ai/common/prompt_fragment.py

```diff
@@ -0,0 +1,31 @@
+from typing import Literal
+
+from pydantic import BaseModel, model_validator
+
+from ai.common.entity_store import EntityStore
+
+
+class PromptFragment(BaseModel):
+  id: str
+  content_value: str | None = None
+  content_path: str | None = None
+  role: Literal["user", "assistant", "system"]
+  chain_of_thought: bool = False
```

Comment:
What does the chain of thought parameter do?

Reply:
it's not really functional right now, but the intention is that if we format a golden dataset and there's a prompt with chain-of-thought, then we include the full LLM output (including the thinking statements) and not just the code/diff.

it's worth experimenting to see if there's any value in this approach, otherwise we can rip it out.